src/PermCheck.php

```diff
@@ -116,13 +116,14 @@ public function run()
                 continue;
             }
 
-            if (!$this->shouldBeExecutable($filename) && $file->isExecutable()) {
+            $fileShouldBeExecutable = $this->shouldBeExecutable($filename);
+            if (!$fileShouldBeExecutable && $file->isExecutable()) {
                 $this->messageBag->addMessage($file->getPathname(), 'minx');
                 $files->next();
                 continue;
             }
 
-            if ($this->shouldBeExecutable($filename) && !$file->isExecutable()) {
+            if ($fileShouldBeExecutable && !$file->isExecutable()) {
```

Comment:
Already changed this up in a different way but that's still on my disk so will pull anyway :)